peerdid/peer_did_utils.py

```diff
@@ -78,25 +86,80 @@ def _create_encnumbasis(key: Union[PublicKeyAgreement, PublicKeyAuthentication])
     return encnumbasis.decode("utf-8")
 
 
-def _decode_encnumbasis(encnumbasis: str, peer_did: PEER_DID) -> dict:
+def _decode_encnumbasis(
+    encnumbasis: str,
+    peer_did: PEER_DID,
+    ver_material_format: VerificationMaterialFormat,
+) -> dict:
     """
     Decodes encnumbasis
     :param encnumbasis: encnumbasis to decode
     :param peer_did: peer_did which will be used as an ID
+    :param ver_material_format: the format of public keys in the DID DOC
     :return: decoded encnumbasis
     """
     decoded_encnumbasis = base58.b58decode(encnumbasis)
-    codec = _get_codec(data=bytes(decoded_encnumbasis))
-    decoded_encnumbasis_without_prefix = _remove_prefix(decoded_encnumbasis)
-    public_key = base58.b58encode(decoded_encnumbasis_without_prefix).decode("utf-8")
+    ver_material = _get_verification_material(decoded_encnumbasis, ver_material_format)
     return {
         "id": peer_did + "#" + encnumbasis,
-        "type": codec,
+        "type": ver_material.type,
         "controller": peer_did,
-        "publicKeyBase58": public_key,
+        ver_material.field: ver_material.value,
     }
 
 
+def _get_verification_material(
+    decoded_encnumbasis: bytes, ver_material_format: VerificationMaterialFormat
+) -> VerificationMaterial:
+    type = _get_public_key_type(data=bytes(decoded_encnumbasis))
+    decoded_encnumbasis_without_prefix = _remove_prefix(decoded_encnumbasis)
+
+    if ver_material_format == VerificationMaterialFormat.BASE58:
+        public_key_value = base58.b58encode(decoded_encnumbasis_without_prefix).decode(
+            "utf-8"
+        )
+        public_key_field = "publicKeyBase58"
+        if type == PublicKeyTypeAgreement.X25519:
+            public_key_type = "X25519KeyAgreementKey2019"
+        elif type == PublicKeyTypeAuthentication.ED25519:
+            public_key_type = "Ed25519VerificationKey2018"
+        else:
+            raise ValueError("Unknown public key type {}".format(type))
+
+    elif ver_material_format == VerificationMaterialFormat.MULTIBASE:
+        public_key_value = "z" + base58.b58encode(
+            decoded_encnumbasis_without_prefix
+        ).decode("utf-8")
+        public_key_field = "publicKeyMultibase"
+        if type == PublicKeyTypeAgreement.X25519:
+            public_key_type = "X25519KeyAgreementKey2019"
+        elif type == PublicKeyTypeAuthentication.ED25519:
+            public_key_type = "Ed25519VerificationKey2020"
+        else:
+            raise ValueError("Unknown public key type {}".format(type))
+
+    elif ver_material_format == VerificationMaterialFormat.JWK:
+        public_key_field = "publicKeyJwk"
+        public_key_type = "JsonWebKey2020"
+        if type == PublicKeyTypeAgreement.X25519 or PublicKeyTypeAuthentication.ED25519:
```

Comment:
I guess something is missed here, possibly `type == PublicKeyTypeAuthentication.ED25519` or `type in (...)`

Reply:
Thank you, fixed